woocommerce/woocommerce-google-analytics-integration
# Update tracking for classic pages

### Changes proposed in this Pull Request:

This is a major restructuring centered around a new [Tracker utility](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/66142e273047f086bf00a780d72bd0b1cb1e5781/assets/js/src/tracker/index.js) which manages all Google Analytics events triggered by either [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/66142e273047f086bf00a780d72bd0b1cb1e5781/assets/js/src/integrations/blocks.js) or [classic pages](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/66142e273047f086bf00a780d72bd0b1cb1e5781/assets/js/src/integrations/classic.js). All tracked events must have an accompanying [data formatter](https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/66142e273047f086bf00a780d72bd0b1cb1e5781/assets/js/src/tracker/data-formatting.js) in order for the Tracker to send the event.

Several files and tests have been removed which are no longer of use 

### Screenshots:

<img width="1108" alt="Screenshot 2024-01-25 at 17 36 37" src="https://github.com/woocommerce/woocommerce-google-analytics-integration/assets/40762232/841287ba-a47e-4e80-9660-8087da1c3d48">

### Test setup

Debug a test website via [Tag Assistant](https://tagassistant.google.com/) that has both Block powered and classic pages for the Cart, Checkout, and Product Listings (Using the `All Products` Block). Ensure all `Event Tracking` options are enabled in `WooCommerce > Settings > Integrations > Google Analytics`.

### Detailed test instructions:

_Work through all test instructions using both types of pages while not logged into the site as an admin._

1. Visit a product archive page and confirm that **one** `view_item_list` event is sent containing all the products on the page
2. Click on a product
3. Confirm that a `select_content` event is sent for that product
4. After the single product loads, confirm that a `view_item` event is sent for the product and a `view_item_list` event is sent if any related products or other lists are displayed
5. Add the product to cart and confirm the `add_to_cart` event is sent
6. Visit the cart, remove the product from the cart and confirm the `remove_from_cart` event it sent
7. Go back to the product archive page and add products to cart from there confirming the event is sent for each
8. Visit the checkout and confirm a `begin_checkout` event is sent
9. Place the order and one the thank you page confirm a `purchase` event is sent
10. Run some basic smoke tests

#### Additional

1. Disable some options in `WooCommerce > Settings > Integrations > Google Analytics`
2. Confirm those events are no longer sent
3. Adjust the `Product Identification` settings
4. Confirm that the data sent with each event includes the correct product identifier structure

⚠️ This branch also includes the changes needed to rename the extension to `Google Analytics for WooCommerce` (_See #353_).

### Changelog entry

> Update - Add shared tracking functionality for WooCommerce Blocks and classic pages
